### ISSUE TITLE
DefaultDict inside DefaultDict is not typed correctly

### ISSUE THREAD

**Issue by kissge** [microsoft/pyright#780]
**Describe the bug**
In the following code, pyright reports inappropriate type error in the line 6:

```
Argument of type "str" cannot be assigned to parameter "k" of type "int" in function "__getitem__"
  "str" is incompatible with "int"Pyright (reportGeneralTypeIssues)
```

**To Reproduce**
```python
from typing import DefaultDict

dic = DefaultDict(lambda: DefaultDict(int))
x: int = 1992
y: str = 'p_km'
dic[x][y] += 928
```

**Expected behavior**
No error.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/4177010/86249402-3ab67980-bbea-11ea-8cb8-2f0234898aea.png)

**VS Code extension or command-line**
VS Code extension (Pyright 1.1.48, Python (extension) 2020.6.91350, VS Code 1.46.1, Python 3.8.3 64-bit)

**Additional context**
n/a

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.49, which I just published.
